go/enclave/storage/enclavedb/block.go

```diff
@@ -216,9 +216,9 @@ func FetchRollupMetadata(ctx context.Context, db *sql.DB, hash common.L2RollupHa
 	return rollup, nil
 }
 
-func fetchBlockHeaders(ctx context.Context, db *sql.DB, whereQuery string, args ...any) ([]*types.Header, error) {
-	result := make([]*types.Header, 0)
-	query := "select header from block " + whereQuery
+func fetchBatchHeaders(ctx context.Context, db *sql.DB, whereQuery string, args ...any) ([]*common.BatchHeader, error) {
```

Comment:
this function already exists ( or something similar) , in batch.go.
Also `FetchBatchHeadersBetween` belongs there as well